datafusion/optimizer/src/single_distinct_to_groupby.rs

```diff
@@ -89,6 +105,24 @@ fn contains_grouping_set(expr: &[Expr]) -> bool {
     matches!(expr.first(), Some(Expr::GroupingSet(_)))
 }
 
+/// Rewrite the aggregate function to two aggregate functions.
+fn aggregate_function_rewrite(
+    fun: &aggregate_function::AggregateFunction,
+) -> (
+    aggregate_function::AggregateFunction,
+    aggregate_function::AggregateFunction,
+) {
+    match fun {
+        Count => (Sum, Count),
```

Comment:
like two phase aggregate, for `Count` aggregate, in first phase we `Count`, and in second phase we `Sum`.